src/view/writer.js

```diff
@@ -15,582 +15,597 @@ import DocumentFragment from './documentfragment.js';
 import isIterable from '../../utils/isiterable.js';
 
 /**
- * Tree View Writer class.
- * Writer defines a high-level API for view manipulations.
+ * Contains functions used for composing view tree.
  *
- * @memberOf engine.view
+ * @namespace engine.view.writer
  */
- export default class Writer {
-	/**
-	 * Returns first parent container of specified {@link engine.view.Position Position}.
-	 * Position's parent node is checked as first, then next parents are checked.
-	 *
-	 * Note that {@link engine.view.DocumentFragment DocumentFragment} is thread like a container.
-	 *
-	 * @param {engine.view.Position} position Position used as a start point to locate parent container.
-	 * @returns {engine.view.ContainerElement|engine.view.DocumentFragment|undefined} Parent container element or
-	 * `undefined` if container is not found.
-	 */
-	getParentContainer( position ) {
-		let parent = position.parent;
-
-		while ( !isContainerOrFragment( parent ) ) {
-			if ( !parent ) {
-				return undefined;
-			}
-			parent = parent.parent;
-		}
 
-		return parent;
-	}
-
-	/**
-	 * Breaks attribute nodes at provided position. It breaks `attribute` nodes inside `container` node.
-	 *
-	 * In following examples `<p>` is a container, `<b>` and `<u>` are attribute nodes:
-	 *
-	 *		<p>foo<b><u>bar{}</u></b></p> -> <p>foo<b><u>bar</u></b>[]</p>
-	 *		<p>foo<b><u>{}bar</u></b></p> -> <p>foo{}<b><u>bar</u></b></p>
-	 *		<p>foo<b><u>b{}ar</u></b></p> -> <p>foo<b><u>b</u></b>[]<b><u>ar</u></b></p>
-	 *
-	 * Note that {@link engine.view.DocumentFragment DocumentFragment} is thread like a container.
-	 *
-	 * @see engine.view.AttributeElement
-	 * @see engine.view.ContainerElement
-	 * @param {engine.view.Position} position Position where to break attributes.
-	 * @returns {engine.view.Position} New position after breaking the attributes.
-	 */
-	breakAttributes( position ) {
-		return this._breakAttributes( position, false );
-	}
-
-	/**
-	 * Private method used by both public breakAttributes (without splitting text nodes) and by other methods (with
-	 * splitting text nodes).
-	 *
-	 * @private
-	 * @param {engine.view.Position} position Position where to break attributes.
-	 * @param {Boolean} [forceSplitText = false] If set to `true`, will break text nodes even if they are directly in
-	 * container element. This behavior will result in incorrect view state, but is needed by other `Writer` methods
-	 * which then fixes view state. Defaults to `false`.
-	 * @returns {engine.view.Position} New position after breaking the attributes.
-	 */
-	_breakAttributes( position, forceSplitText = false ) {
-		const positionOffset = position.offset;
-		const positionParent = position.parent;
-
-		// There are no attributes to break and text nodes breaking is not forced.
-		if ( !forceSplitText && positionParent instanceof Text && isContainerOrFragment( positionParent.parent ) ) {
-			return Position.createFromPosition( position );
-		}
+export default {
+	getParentContainer,
```

Comment:
In similar way, `breakRange`, `breakAttributes` and `mergeAttributes` looks more like helper methods for `viewWriter`. Similar functions will be in `modelWriter` (after #480) and are now in `NodeList` - also private.

When creating features, developers should not care about things like breaking and merging nodes. They should just use `insert`, `wrap`, `move`, etc. Unfortunately I can see that `breakAttributes` is actually used in `model-selection-to-view-converters.js` which bothers me.

Except of selection conversion I can't imagine why someone would like to just break attributes, normally you would like to break attributes because you need to `insert` something or `remove` something, etc.

Since there is no quick way around this, we have to keep `breakAttributes` and (for consistency) `breakRange`, `mergeAttributes` exposed.